fetch.bs

```diff
@@ -3265,6 +3276,109 @@ through TLS using ALPN. The protocol cannot be spoofed through HTTP requests in
 </div>
 
 
+<h3 id=https-upgrades>HTTPS upgrading</h3>
+
+<p>User agents may optionally upgrade requests with URLs that are not
+<a>potentially trustworthy URLs</a> to attempt to fetch them over
+<a>potentially trustworthy URLs</a>. If an upgraded request fails with a network error, it is
+retried over the original URL.
+
+<p>The HTTPS upgrading algorithm consists of <a>upgrade an HTTP request</a> and
+<a>HTTPS upgrade fallback</a> algorithms.
+
+<h4 id=https-upgrades-upgrade>HTTPS upgrade algorithm</h4>
+<div algorithm>
+
+<p>To <dfn>upgrade an HTTP request</dfn> given a <a for=/>request</a> <var>request</var>:
+
+<ol>
+ <li>
+  <p>If one or more of the following conditions are met, return:
+  <ul>
```

Comment:
[Mixed Content's upgrade logic](https://w3c.github.io/webappsec-mixed-content/#upgrade-algorithm) has an early return for `request's URL's host is an IP address`.

Was this considered here?

Reply:
The main difference is that Mixed content doesn't have a fallback path. So while upgrading an IP address is unlikely to work in most cases, it might still work, so it's still worth doing an HTTPS Upgrade for those cases. A failed load will always fallback to the original request, so upgrading an IP address should be safe, as opposed to mixed content.

(I'm happy to add a note to that effect here, if you prefer)

Reply:
I thought you can not get a certificate for an IP address, but looks like that was wrong.

In that case, would it make sense to exempt local IP addresses then?

Reply:
I think development/testing is still a valid use case for upgrading local IP addresses though? UAs can remember the fallback event and not upgrade again for N days, a single one time upgrade is still worth trying for these. Chrome currently allowlists the hostname for 7 days and extends the allowlist entry on every navigation, so the user would effectively encounter a single fallback at the beginning, as long as they keep navigating to the same URL.